app/controllers/dashboards_controller.rb

```diff
@@ -1,6 +1,6 @@
 class DashboardsController < ApplicationController
-  before_action :authenticate_with_api_key, unless: :signed_in?
-  before_action :redirect_to_signin, unless: -> { signed_in? || @api_user }
+  before_action :find_api_key, unless: :signed_in?
```

Comment:
As above, consider renaming `find_api_key` to clarify it performs authentication.